src/StakeRegistryStorage.sol

```diff
@@ -46,9 +50,15 @@ abstract contract StakeRegistryStorage is IStakeRegistry {
 
 
     constructor(
+        IServiceManager _serviceManager,
         IRegistryCoordinator _registryCoordinator, 
         IDelegationManager _delegationManager
     ) {
+        require(
+            _registryCoordinator.serviceManager() == _serviceManager,
```

Comment:
we don't do this anywhere else.

either we should not add the param and just call `.serviceManager` or dont check the param imo

Reply:
Oh yeah could just use .serviceManager 
Will remove param

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/263/commits/9880a8de41a12856928e756dd244d1f4b16c75b9#diff-3aca207bfe24bd32c66e23f704da75c45051c4fad2b745aa2efaf937e9dc6e2d